api/test/wfl/system/cdc_covid19_surveillance_demo.clj

```diff
@@ -0,0 +1,125 @@
+    clone-name))
+
+(defn look-for-new-dataset-rows [{:keys [name] :as dataset}]
+  (println "Looking for rows to snapshot in dataset" name)
+  (let [row-ids (-> (datarepo/query-table-between
+                     dataset source-table
+                     snapshot-column
+                     ["2021-04-24 03:00:00" "2021-04-26 05:00:00"]
+                     [:datarepo_row_id])
+                    :rows
+                    flatten)]
+    (println "Found rows" row-ids)
+    row-ids))
+
+(defn snapshot-new-dataset-rows [{:keys [name] :as dataset} row-ids]
+  (println "Snapshotting rows" row-ids "in dataset" name)
+  (let [{:keys [name] :as snapshot}
+        (-> (snapshots/unique-snapshot-request
+             source-dataset-profile
+             dataset
+             source-table
+             row-ids)
+            (assoc :readers ["workflow-launcher-dev@firecloud.org"])
+            datarepo/create-snapshot
+            datarepo/snapshot)]
+    (println "Created snapshot" name)
+    snapshot))
+
+(defn create-snapshot []
+  (let [dataset (datarepo/dataset source-dataset)]
+    (snapshot-new-dataset-rows dataset (look-for-new-dataset-rows dataset))))
+
+(defn import-snapshot-into-workspace [workspace {:keys [name id] :as _snapshot}]
+  (println "Importing snapshot" name "into" workspace)
+  (let [{:keys [name] :as ref} (rawls/create-snapshot-reference workspace id name)]
+    (println "Created snapshot reference" name)
+    ref))
+
+(defn update-method-configuration
+  [workspace {:keys [name] :as _snapshot-reference}]
+  (println "Updating" method-configuration "to use" name)
+  (-> (firecloud/get-method-configuration workspace method-configuration)
+      (assoc :dataReferenceName name)
+      (->> (firecloud/update-method-configuration workspace method-configuration))))
+
+(defn submit-snapshot-reference
+  [workspace {:keys [name] :as _snapshot-reference}]
+  (println "Submitting" name "in" workspace)
+  (let [submission (firecloud/submit-method workspace method-configuration)
+        [id count] ((juxt :submissionId (comp count :workflows)) submission)]
+    (println "Created submission" id "-" count "workflow(s)")
+    submission))
+
+(defn abort-submission [workspace {:keys [submissionId] :as _submission}]
+  (println "Aborting" submissionId "in" workspace)
+  (firecloud/abort-submission workspace submissionId))
+
+(defn write-known-outputs-to-workspace [workspace]
+  (println "Writing outputs to flowcell table in" workspace))
+
+(defn delete-snapshot [{:keys [name id] :as _snapshot}]
+  (println "Deleting snapshot" name)
+  (datarepo/delete-snapshot id))
+
+(defn delete-workspace [workspace]
+  (println "Deleting" workspace)
+  (firecloud/delete-workspace workspace))
+
+(defn demo []
+  (println "CDC COVID-19 Surveillance Demo")
+  (println "Demonstrating sarscov2_illimina_full automation")
+  (wait-for-user)
+  (fixtures/with-fixtures
+    [(util/bracket clone-workspace delete-workspace)
```

Comment:
I'm guessing that if we delete a workspace, everything within it (snapshot references, methodconfigs, submissions) are also deleted.

Reply:
Yep